# **Business Understanding**

**About Dataset**
This dataset contains used car listings sourced from Craigslist (United States) that have been cleaned to remove low-value or redundant columns and to make the data more usable for analysis and predictive modeling. The cleaned data is often used for exploratory data analysis, price prediction models, and market trend studies

[Source](https://www.kaggle.com/datasets/youssefkhaled117/used-car-cleaned)

### **Loading libraries** 


In [2]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt

### **Loading data** 

In [3]:
used_cars_df = pd.read_csv('data/Used_Cars_Cleaned.csv')

# **Data Understanding**
- head/tail of the data
- shape
- column
- info
- data types
- missing values
- outliers
- duplicates
- descriptive stats(quantitative & qualitative)
- basic visualizations


#### **Get a glimpse of the data(head, tail)**

In [4]:
used_cars_df.head(10)

,id,region,price,year,manufacturer,model,condition,cylinders,fuel,odometer,title_status,transmission,drive,type,paint_color,state,posting_date,posting_month
0,7222695916,prescott,6000,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,az,2021-04-23,4
1,7218891961,fayetteville,11900,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,ar,2021-04-23,4
2,7221797935,florida keys,21000,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,fl,2021-04-23,4
3,7222270760,worcester / central MA,1500,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,ma,2021-04-23,4
4,7210384030,greensboro,4900,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,nc,2021-04-23,4
5,7222379453,hudson valley,1600,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,ny,2021-04-23,4
6,7221952215,hudson valley,1000,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,ny,2021-04-23,4
7,7220195662,hudson valley,15995,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,ny,2021-04-23,4
8,7209064557,medford-ashland,5000,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,or,2021-04-23,4
9,7219485069,erie,3000,2017.0,ford,f-150,good,8 cylinders,gas,102505.0,clean,automatic,4wd,pickup,white,pa,2021-04-23,4


In [5]:
used_cars_df.tail(10)

,id,region,price,year,manufacturer,model,condition,cylinders,fuel,odometer,title_status,transmission,drive,type,paint_color,state,posting_date,posting_month
392002,7301592119,wyoming,22990,2020.0,hyundai,sonata se sedan 4d,good,4 cylinders,gas,3066.0,clean,other,fwd,sedan,blue,wy,2021-04-04,4
392003,7301591639,wyoming,17990,2018.0,kia,sportage lx sport utility 4d,good,6 cylinders,gas,34239.0,clean,other,fwd,SUV,silver,wy,2021-04-04,4
392004,7301591201,wyoming,32590,2020.0,mercedes-benz,c-class c 300,good,4 cylinders,gas,19059.0,clean,other,rwd,sedan,white,wy,2021-04-04,4
392005,7301591202,wyoming,30990,2018.0,mercedes-benz,glc 300 sport,good,6 cylinders,gas,15080.0,clean,automatic,rwd,other,white,wy,2021-04-04,4
392006,7301591199,wyoming,33590,2018.0,lexus,gs 350 sedan 4d,good,6 cylinders,gas,30814.0,clean,automatic,rwd,sedan,white,wy,2021-04-04,4
392007,7301591192,wyoming,23590,2019.0,nissan,maxima s sedan 4d,good,6 cylinders,gas,32226.0,clean,other,fwd,sedan,blue,wy,2021-04-04,4
392008,7301591187,wyoming,30590,2020.0,volvo,s60 t5 momentum sedan 4d,good,6 cylinders,gas,12029.0,clean,other,fwd,sedan,red,wy,2021-04-04,4
392009,7301591147,wyoming,34990,2020.0,cadillac,xt4 sport suv 4d,good,6 cylinders,diesel,4174.0,clean,other,fwd,hatchback,white,wy,2021-04-04,4
392010,7301591140,wyoming,28990,2018.0,lexus,es 350 sedan 4d,good,6 cylinders,gas,30112.0,clean,other,fwd,sedan,silver,wy,2021-04-04,4
392011,7301591129,wyoming,30590,2019.0,bmw,4 series 430i gran coupe,good,6 cylinders,gas,22716.0,clean,other,rwd,coupe,black,wy,2021-04-04,4


#### **Shape/ dimensions & columns/ features of data**

In [6]:
rows, columns  = used_cars_df.shape
output = f"On craigs lists we have {rows} records/listings and {columns} features/attributes unser the used cars."
print(output)

On craigs lists we have 392012 records/listings and 18 features/attributes unser the used cars.


In [7]:
used_cars_df.columns

Index(['id', 'region', 'price', 'year', 'manufacturer', 'model', 'condition',
       'cylinders', 'fuel', 'odometer', 'title_status', 'transmission',
       'drive', 'type', 'paint_color', 'state', 'posting_date',
       'posting_month'],
      dtype='str')

#### **Data types, duplicates & info**

In [8]:
used_cars_df.dtypes

id                 int64
region               str
price              int64
year             float64
manufacturer         str
model                str
condition            str
cylinders            str
fuel                 str
odometer         float64
title_status         str
transmission         str
drive                str
type                 str
paint_color          str
state                str
posting_date         str
posting_month      int64
dtype: object

In [9]:
used_cars_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 392012 entries, 0 to 392011
Data columns (total 18 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   id             392012 non-null  int64  
 1   region         392012 non-null  str    
 2   price          392012 non-null  int64  
 3   year           392012 non-null  float64
 4   manufacturer   392012 non-null  str    
 5   model          392012 non-null  str    
 6   condition      392012 non-null  str    
 7   cylinders      392012 non-null  str    
 8   fuel           392012 non-null  str    
 9   odometer       392012 non-null  float64
 10  title_status   392012 non-null  str    
 11  transmission   392012 non-null  str    
 12  drive          392012 non-null  str    
 13  type           392012 non-null  str    
 14  paint_color    392012 non-null  str    
 15  state          392012 non-null  str    
 16  posting_date   392012 non-null  str    
 17  posting_month  392012 non-null  int64  


In [11]:
used_cars_df.duplicated().sum()

np.int64(0)

#### **Insight:**
- On craigs lists we have 392012 records/listings and 18 features/attributes unser the used cars
- No missing values and No duplicate data
- 5 quantitative featues and 13 qualitative features



#### **Descriptive Stats (quantitative and qualitative)**